modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/TableRowConverterImpl.java

```diff
@@ -46,66 +41,36 @@ public class TableRowConverterImpl implements TableRowConverter {
 
     private final BinaryTupleSchema fullTupleSchema;
 
-    private final int[] fullRowColocationIndexes;
-    private final int[] keyOnlyColocationIndexes;
-    private final NativeType[] colocationColumnTypes;
-
     /**
      * Mapping of required columns to their indexes in physical schema.
      */
     private final int[] requiredColumnsMapping;
 
-    /**
-     * Mapping of all columns to their indexes in physical schema.
-     */
-    private final int[] fullMapping;
-
-    private final boolean skipReshuffling;
+    private final boolean skipTrimming;
 
     /** Constructor. */
     TableRowConverterImpl(
             SchemaRegistry schemaRegistry,
             BinaryTupleSchema fullTupleSchema,
-            int[] fullRowColocationIndexes,
-            int[] keyOnlyColocationIndexes,
-            NativeType[] colocationColumnTypes,
             SchemaDescriptor schemaDescriptor,
             @Nullable BitSet requiredColumns
     ) {
         this.schemaRegistry = schemaRegistry;
         this.schemaDescriptor = schemaDescriptor;
-        this.fullRowColocationIndexes = fullRowColocationIndexes;
-        this.keyOnlyColocationIndexes = keyOnlyColocationIndexes;
-        this.colocationColumnTypes = colocationColumnTypes;
         this.fullTupleSchema = fullTupleSchema;
 
-        this.skipReshuffling = requiredColumns == null && schemaDescriptor.physicalOrderMatchesLogical();
+        this.skipTrimming = requiredColumns == null;
```

Comment:
Maybe it is better make the class package-private and have 2 constructor for both cases: `requiredColumns` is null and not null; instead of having bunch of `if (requiredColumns == null)`? Or maybe different implementations?
In fact, the object is always created via factory and it is clear which constructor must be used.

Reply:
good point! Filed an issue https://issues.apache.org/jira/browse/IGNITE-21732